transport/awssqs/publisher.go

```diff
@@ -0,0 +1,126 @@
+package awssqs
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/go-kit/kit/endpoint"
+)
+
+// Publisher wraps an sqs client and queue, and provides a method that
+// implements endpoint.Endpoint.
+type Publisher struct {
+	sqsClient        Client
+	queueURL         string
+	responseQueueURL string
```

Comment:
What the sense of this parameter? Is it not a specific use case when you want to provide your "responseQueueURL" attribute? Someone can want to use "responseUrl" and then would be a problem. 
If you want to use your custom messageAttributes, then you should use it in "before" or in encodeMessage, but not in the general approach.

Reply:
The Publisher is the one that sets the responseQueue for his messages.
As the SQS doc [response reply pattern implementation](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/working-with-messages.html#implementing-request-response-systems) states :

> Don't create reply queues per message. Instead, create reply queues on startup, per producer, and use a correlation ID message attribute to map replies to requests.

So this field actually makes sense in the Producer.

The Consumer on the other hand will indeed retrieve this value from a message attribute that will be added in the message in a before function.

Reply:
Ok, I understand your desire for this pattern, but why should we use this "responseQueueURL" as the field of Publisher? I  don't see any uses of this field, so then if you want to add this attribute, you will add in your EncodeRequestFunc or in PublisherBefore option.

Reply:
Indeed the field was not used, I just added it as paramter to the Publisher's before function